lib/block-supports/layout.php

```diff
@@ -28,14 +28,15 @@ function gutenberg_register_layout_support( $block_type ) {
 /**
  * Generates the CSS corresponding to the provided layout.
  *
- * @param string  $selector CSS selector.
- * @param array   $layout   Layout object. The one that is passed has already checked the existence of default block layout.
- * @param boolean $has_block_gap_support Whether the theme has support for the block gap.
- * @param string  $gap_value The block gap value to apply.
+ * @param string  $selector                      CSS selector.
+ * @param array   $layout                        Layout object. The one that is passed has already checked the existence of default block layout.
+ * @param boolean $has_block_gap_support         Whether the theme has support for the block gap.
+ * @param string  $gap_value                     The block gap value to apply.
+ * @param boolean $should_skip_gap_serialization Whether to skip applying the user-defined value set in the editor.
```

Comment:
Should we not also allow skipping serialization for the whole layout feature too? The Navigation block already implements [custom logic](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/navigation/edit/index.js#L331) to apply the layout styles to its inner elements and blocks, so there's definitely a use case.

Reply:
> Should we not also allow skipping serialization for the whole layout feature too?

Good point! Thanks for mentioning it. I think this would be a good follow up PR. 

I'm guessing @aaronrobertshaw started the work here with a view of extending block supports (border, spacing et. al) that already use `skipSerialization` in various block.jsons today.

The layout implementation, being a little "different" might also need a level of testing that might be more suited to a dedicated PR. 😄

Reply:
Yup, layout is complex enough that a follow-up PR sounds right.

Reply:
I'm only just getting up to speed after some recent AFK so might be missing some context.

When originally creating this I found a few block supports that simply didn't maintain/add the ability to skip serialization as per other block supports. Layout was one example of this. It also changed several times during the long life span of this PR. Early on, I had a skip for the entire layout feature but the support was reworked which, from memory, removed the code where that could easily be performed.

I think a separate PR would definitely be a good idea given how complicated layout support is and how much it's already changed in its relatively short life.